contracts/ERC1155LaunchpegBase.sol

```diff
@@ -0,0 +1,357 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import {AccessControlEnumerableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/AccessControlEnumerableUpgradeable.sol";
+import {ERC1155Upgradeable} from "@openzeppelin/contracts-upgradeable/token/ERC1155/ERC1155Upgradeable.sol";
+import {ERC2981Upgradeable} from "@openzeppelin/contracts-upgradeable/token/common/ERC2981Upgradeable.sol";
+import {ReentrancyGuardUpgradeable} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {StringsUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/StringsUpgradeable.sol";
+import {IOperatorFilterRegistry} from "operator-filter-registry/src/IOperatorFilterRegistry.sol";
+
+import "./LaunchpegErrors.sol";
+import {SafePausableUpgradeable} from "./utils/SafePausableUpgradeable.sol";
+
+import {IERC1155LaunchpegBase} from "./interfaces/IERC1155LaunchpegBase.sol";
+
+abstract contract ERC1155LaunchpegBase is
+    IERC1155LaunchpegBase,
+    ERC1155Upgradeable,
+    ERC2981Upgradeable,
+    ReentrancyGuardUpgradeable,
+    SafePausableUpgradeable
+{
+    using StringsUpgradeable for uint256;
+
+    /// @notice Percentage base point
+    uint256 private constant BASIS_POINT_PRECISION = 10_000;
+
+    /// @notice Role granted to project owners
+    bytes32 public constant PROJECT_OWNER_ROLE =
+        keccak256("PROJECT_OWNER_ROLE");
+
+    /// @notice Contract filtering allowed operators, preventing unauthorized contract to transfer NFTs
+    /// By default, Launchpeg contracts are subscribed to OpenSea's Curated Subscription Address at 0x3cc6CddA760b79bAfa08dF41ECFA224f810dCeB6
+
+    IOperatorFilterRegistry public operatorFilterRegistry;
+
+    /// @notice The fees collected by Joepegs on the sale benefits
+    /// @dev In basis points e.g 100 for 1%
+    uint256 public joeFeePercent;
+
+    /// @notice The address to which the fees on the sale will be sent
+    address public joeFeeCollector;
+
+    /// @notice Start time when funds can be withdrawn
+    uint256 public withdrawAVAXStartTime;
```

Comment:
This could be updated to:
```solidity
    /// @notice The address to which the fees on the sale will be sent
    address public joeFeeCollector;

    /// @notice The fees collected by Joepegs on the sale benefits
    /// @dev In basis points e.g 100 for 1%
    uint8 public joeFeePercent;

    /// @notice Start time when funds can be withdrawn
    uint80 public withdrawAVAXStartTime;

    /// @notice This boolean can be turned on to prevent any changes on the sale parameters.
    /// @dev Once set to true, it shouldn't be possible to turn it back to false.
    bool public locked;
```

To save some gas. Don't forget to use safe casting when casting values if they're not safe to cast directly